### PR TITLE
Add ContentHash struct, replace TransactionalContentMD5 and TransactionContentCrc64 with TransactionalContentHash

### PR DESCRIPTION
**Pull request by Jinming-Hu**
# Pull Request Checklist

Please leverage this checklist as a reminder to address commonly occurring feedback when submitting a pull request to make sure your PR can be reviewed quickly:

See the detailed list in the [contributing guide](https://github.com/Azure/azure-sdk-for-cpp/blob/master/CONTRIBUTING.md#pull-requests).

- [x] [C++ Guidelines](https://azure.github.io/azure-sdk/cpp_introduction.html)
- [x] Doxygen docs
- [x] Unit tests
- [x] No unwanted commits/changes
- [x] Descriptive title/description
  - [x] PR is single purpose
  - [x] Related issue listed
- [x] Comments in source
- [x] No typos
- [x] Update changelog
- [x] Not work-in-progress
- [x] External references or docs updated
- [x] Self review of PR done
- [x] Any breaking changes?

Split https://github.com/Azure/azure-sdk-for-cpp/pull/1111 into two separate PRs, one for Blob service and one for DataLake/File service. So that other changes to blob protocol layer is unblocked

### CODE REVIEW

File: sdk/storage/azure-storage-blobs/inc/azure/storage/blobs/blob_options.hpp [hunk 1]
```diff
@@ -106,9 +106,9 @@ namespace Azure { namespace Storage { namespace Blobs {
     std::string Key;
 
     /**
-     * @brief Base64 encoded string of the AES256 encryption key's SHA256 hash.
+     * @brief SHA256 hash of the AES256 encryption key.
      */
-    std::string KeyHash;
```
Comment (katmsft) [inline sdk/storage/azure-storage-blobs/inc/azure/storage/blobs/blob_options.hpp:111]: This change is not mentioned in change log.

File: sdk/storage/azure-storage-blobs/test/block_blob_client_test.cpp [hunk 2]
```diff
@@ -45,7 +45,7 @@ namespace Azure { namespace Storage { namespace Test {
     m_blobUploadOptions.HttpHeaders.ContentDisposition = "attachment";
     m_blobUploadOptions.HttpHeaders.CacheControl = "no-cache";
     m_blobUploadOptions.HttpHeaders.ContentEncoding = "identity";
-    m_blobUploadOptions.HttpHeaders.ContentMd5 = "";
```
Comment (katmsft) [inline sdk/storage/azure-storage-blobs/test/block_blob_client_test.cpp:48]: Should HttpHeaders be containing `ContentHash` or `ContentMd5`?

Comment (Jinming-Hu) [reply sdk/storage/azure-storage-blobs/test/block_blob_client_test.cpp:48]: The type is `std::vector<uint8_t>` so I think it should be `ContentMd5`

Comment (Jinming-Hu) [reply sdk/storage/azure-storage-blobs/test/block_blob_client_test.cpp:48]: We agreed on this yesterday

File: sdk/storage/azure-storage-common/CMakeLists.txt [hunk 3]
```diff
@@ -75,17 +75,18 @@ target_include_directories(
       $<INSTALL_INTERFACE:include>
 )
 
+# libxml2 headers and libraries will be changed to PRIVATE in the future.
 target_include_directories(azure-storage-common PUBLIC ${LIBXML2_INCLUDE_DIRS})
 target_link_libraries(azure-storage-common PUBLIC Azure::azure-core)
 target_link_libraries(azure-storage-common INTERFACE Threads::Threads ${LIBXML2_LIBRARIES})
 
 if(MSVC)
-    target_link_libraries(azure-storage-common INTERFACE bcrypt)
```
Comment (katmsft) [inline sdk/storage/azure-storage-common/CMakeLists.txt:83]: Why is this change related?

Comment (Jinming-Hu) [reply sdk/storage/azure-storage-common/CMakeLists.txt:83]: It's not that related.

File: sdk/storage/azure-storage-blobs/inc/azure/storage/blobs/blob_options.hpp (hunk 1 above)
Comment (Jinming-Hu) [reply sdk/storage/azure-storage-blobs/inc/azure/storage/blobs/blob_options.hpp:111]: I was thinking, shouldn't this also be `ContentHash`, with the HashAlgorithm bing `Sha256`?

File: sdk/storage/azure-storage-blobs/test/block_blob_client_test.cpp (hunk 2 above)
Comment (Jinming-Hu) [reply sdk/storage/azure-storage-blobs/test/block_blob_client_test.cpp:48]: https://github.com/Azure/azure-sdk-for-cpp/issues/1213

**Review by katmsft — APPROVED** [review]
Approve with a follow-up item
